### PR TITLE
Vmware provider

### PR DESCRIPTION
**Pull request by CDimonaco**
# Description

Add `vmware` as provider.

This pr updates both the backend/frontend with relevant tests

## How was this tested?

Automated tests

### CODE REVIEW

File: assets/js/components/ProviderLabel/ProviderLabel.jsx [hunk 1]
```diff
@@ -29,6 +30,10 @@ export const providerData = {
     logo: KvmLogo,
     label: 'KVM',
   },
+  vmware: {
+    logo: VmwareLogo,
+    label: 'Vmware',
```
Comment (arbulu89) [inline assets/js/components/ProviderLabel/ProviderLabel.jsx:35]: The text should be `VMware` as it is the official way (`M` in upper case)

File: assets/js/components/ProviderLabel/ProviderLabel.test.jsx [hunk 2]
```diff
@@ -37,6 +37,14 @@ describe('Provider Label', () => {
       'nutanix'
     );
   });
+
+  it('should display an icon and label with VMWare as the provider', () => {
```
Comment (arbulu89) [inline assets/js/components/ProviderLabel/ProviderLabel.test.jsx:41]: Better `VMware`

**Review by arbulu89 — APPROVED** [review]
Nice!
Some nitpick. 
- `Vmware` -> `VMware`.
- We could add `vmware` to the `assets/js/lib/test-utils/factories/hosts.js` factory (kvm as well is missing)
- Not about this PR, but some openapi schema fields are stucked with `enum: [:azure, :aws, :gcp, :default]` which looks outdated


PD: It would nice to have some screenshots to see the icons and etc

File: assets/js/components/HostDetails/ProviderDetails.test.jsx [hunk 3]
```diff
@@ -21,6 +21,10 @@ describe('Provider Details', () => {
       provider: 'kvm',
       providerText: 'KVM',
     },
+    {
+      provider: 'vmware',
+      providerText: 'Vmware',
```
Comment (rtorrero) [inline assets/js/components/HostDetails/ProviderDetails.test.jsx:26]: Tiny absurd detail but it'd be nice if the capitalization here is VMware.

**Review by rtorrero — APPROVED** [review]
Looks good, just a minor comment about capitalization. I'd change every user facing labels for vmware to "VMware"

EDIT: already mentioned by @arbulu89

**Comment by CDimonaco** [comment]
Addressed all comments, screenshots incoming

**Comment by CDimonaco** [comment]
image
image

File: assets/js/components/ProviderLabel/ProviderLabel.test.jsx (hunk 2 above)
Comment (fabriziosestito) [reply assets/js/components/ProviderLabel/ProviderLabel.test.jsx:41]: nitpick: `VMWare -> VMware`

**Review by fabriziosestito — APPROVED** [review]
LGTM!
we need to agree on what cypress testing should test, but for the moment adding these two tests doesn't hurt

**Pull request merged**